File: Baekjoon_Online_Judge/NO_14914/14914.cpp
#include <iostream>
#include <cmath>
using namespace std;
/*
 ---------------------------- 
 | Created by happiness96   | 
 | Year 2020                | 
 | Month 07                 | 
 | Day 28                   | 
 | 14914 사과와 바나나 나눠주기 | 
 ---------------------------- 
*/

int main(){
    int a, b;

    cin >> a >> b;

    for(int i = 1; i <= min(a, b); i ++){
        if(a % i == 0 and b % i == 0){
            cout << i << ' ' << a / i << ' ' << b / i << endl;
        }
    }

    return 0;
}

File: Baekjoon_Online_Judge/NO_14914/14914.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# ---------------------------- #
# | Created by happiness96   | #
# | Year 2020                | #
# | Month 07                 | #
# | Day 28                   | #
# | 14914 사과와 바나나 나눠주기 | #
# ---------------------------- #


if __name__ == "__main__":
    a, b = map(int, r_input().split())

    for cnt in range(1, min(a, b) + 1):
        if a % cnt == b % cnt == 0:
            print(cnt, a // cnt, b // cnt)
